### PR TITLE
GEODE-6958: Improve JMH integration

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
* Generate correct JMH shadow jar for standalone execution.
* Fix plugin dependencies.
* Add profiler command line property.
* Adds README

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -49,7 +49,7 @@ buildId = 0
 productName = Apache Geode
 productOrg = Apache Software Foundation (ASF)
 
-minimumGradleVersion = 5.4
+minimumGradleVersion = 5.5.1
```
Comment (rhoughton-pivot) [inline gradle.properties]: Why the bump to Gradle 5.5.1 if we aren't also moving to the latest JMH plugin?

File: geode-core/build.gradle [hunk 2]
```diff
@@ -417,6 +407,28 @@ dependencies {
...
 
+jmh {
+  duplicateClassesStrategy = 'warn'
+  include = project.hasProperty('jmh.include') ? project.getProperties().get('jmh.include') : '.*'
+  profilers = project.hasProperty('jmh.profilers') ? project.getProperties().get('jmh.profilers').tokenize() : []
+  threads = project.hasProperty('jmh.threads') ? project.getProperties().get('jmh.threads') : null
+}
+
+jmhJar {
+  from {
```
Comment (rhoughton-pivot) [inline geode-core/build.gradle]: Does this change away from using a sub-task for collecting the files make the UP-TO-DATE check work as   expected?

File: gradle.properties (hunk 1 above)
Comment (pivotal-jbarrett) [reply gradle.properties]: I started to update the plugin here but then solved it another way. I can back out this Gradle upgrade here and open another PR.

File: geode-core/build.gradle (hunk 2 above)
Comment (pivotal-jbarrett) [reply geode-core/build.gradle]: Yes, but let me double check it still works.

**Pull request merged**